src/scheduler/Scheduler.cpp

```diff
@@ -486,7 +498,12 @@ faabric::util::SchedulingDecision Scheduler::doSchedulingDecision(
     }
 
     // Sanity check
-    assert(hosts.size() == nMessages);
+    if (hosts.size() != nMessages) {
+        SPDLOG_ERROR(
+          "Serious scheduling error: {} != {}", hosts.size(), nMessages);
+
+        throw std::runtime_error("Not enough scheduled hosts for messages");
```

Comment:
This should never happen, but given that it did happen, I found it much nicer to have a message than an assertion error.